components/bl-autocomplete-component/src/helpers/use-on-click-outside.js

```diff
@@ -0,0 +1,21 @@
+import { useEffect } from 'react';
+
+export const useOnClickOutside = (ref, handler) => {
+  const listener = event => {
+    if (!ref.current || ref.current.contains(event.target)) {
+      return;
+    }
+
+    handler(event);
+  };
+
+  useEffect(() => {
+    document.addEventListener('mousedown', listener);
+    document.addEventListener('touchstart', listener);
+
+    return () => {
+      document.removeEventListener('mousedown', listener);
+      document.removeEventListener('touchstart', listener);
+    };
+  }, [ref, handler]);
```

Comment:
- `ref` won't be changed because it's a Ref, I assume
- I do not see where you use `handler` in the hook
- should not we add `listener` to the deps?

```
 
  useEffect(() => {
 function listener(event) {
    if (ref.current && !ref.current.contains(event.target)) {
      handler(event);
    }
  };

    document.addEventListener('mousedown', listener);
    document.addEventListener('touchstart', listener);

    return () => {
      document.removeEventListener('mousedown', listener);
      document.removeEventListener('touchstart', listener);
    };
  }, [ref.current, handler]);
```

?